app/src/main/java/com/example/starwars_kotlin/di/ApiModule.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.starwars_kotlin.di
+
+import androidx.lifecycle.MutableLiveData
+import com.example.starwars_kotlin.adapter.PeopleAdapter
+import com.example.starwars_kotlin.api_endpoint.StarWarsApi
+import com.example.starwars_kotlin.model.People
+import com.example.starwars_kotlin.service.NetworkService
+import dagger.Module
+import dagger.Provides
+import io.reactivex.disposables.CompositeDisposable
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+
+@Module
+class ApiModule {
+
+    @Provides
+    fun provideStarWarsApi(): StarWarsApi {
+        return Retrofit.Builder()
+            .baseUrl(NetworkService.BASE_URL)
+            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+            .addConverterFactory(MoshiConverterFactory.create())
+            .build()
+            .create(StarWarsApi::class.java)
+    }
```

Comment:
Você poderia fazer esse retorno em uma função estática dentro da interface do `StarWarsApi` como eu fiz abaixo:
```
interface StarWarsApi {
    ...
    companion object {
        val instance: StarWarsApi
            get() =  Retrofit.Builder()
                .baseUrl(NetworkService.BASE_URL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(MoshiConverterFactory.create())
                .build()
                .create(StarWarsApi::class.java)
    }
} 
```
Porque da forma que você está fazendo, você tá acoplando o código de instanciação da `StarWarsApi` em outros lugares, e seria melhor ela mesma saber se instanciar. Com isso, o seu código na função `provideStarWarsApi()` iria ficar:
```suggestion
    @Provides
    fun provideStarWarsApi() = StarWarsApi.instance
```